core/arch/arm/kernel/thread.c

```diff
@@ -1461,11 +1461,29 @@ static void set_ctx_regs(struct thread_ctx_regs *regs, unsigned long a0,
 	regs->cpsr = spsr;
 	regs->x[13] = user_sp;	/* Used when running TA in Aarch32 */
 	regs->sp = user_sp;	/* Used when running TA in Aarch64 */
+#ifdef CFG_TA_PAUTH
+	assert(keys);
+	regs->apiakey_hi = keys->hi;
+	regs->apiakey_lo = keys->lo;
+#endif
 	/* Set frame pointer (user stack can't be unwound past this point) */
 	regs->x[29] = 0;
 #endif
 }
 
+static struct thread_pauth_keys *thread_get_pauth_keys(void)
+{
+#ifdef CFG_TA_PAUTH
```

Comment:
if think using `if (IS_ENABLED(CFG_TA_PAUTH)) return NULL;` should build ok.

Reply:
This doesn't build fine as struct user_mode_ctx’ has no member named ‘keys’ if CFG_TA_PAUTh is not enabled.